RELEASES.md

```diff
@@ -20,6 +54,63 @@ Unreleased.
   than closing the connection.
   [#10645](https://github.com/bytecodealliance/wasmtime/pull/10645)
 
+* Cranelift's single-pass allocator has been disabled due to being unable to
+  support internal refactorings in preparation for the WebAssembly exceptions
+  proposal. Re-enabling this allocator is tracked at
+  [regalloc2#217](https://github.com/bytecodealliance/regalloc2/issues/217) for
+  those interested.
+  [#10554](https://github.com/bytecodealliance/wasmtime/pull/10554)
+
+* Wasmtime's `{Array,Extern,Struct}Ref` functions will now automatically trigger
+  a GC.
+  [#10560](https://github.com/bytecodealliance/wasmtime/pull/10560)
+
+* Wasmtime's GC heaps now use the same translation techniques as linear memories
+  meaning they have far fewer bounds-checks than before.
+  [#10503](https://github.com/bytecodealliance/wasmtime/pull/10503)
+
+* Wasmtime's implementation of WASIp2 has moved to `wasmtime_wasi::p2` from the
+  root of the crate.
+  [#10073](https://github.com/bytecodealliance/wasmtime/pull/10073)
+
+* Wasmtime will no longer emit calls to Cranelift-defined "libcalls" and instead
+  everything goes through Wasmtime's libcall mechanism instead, paving the way
+  for a future change for more efficient stack limit checking in wasm.
+  [#10657](https://github.com/bytecodealliance/wasmtime/pull/10657)
```

Comment:
should we note here that this also removes the need to perform relocations on the code, improving load times and allowing better use of cached pages when mapping from disk?

Reply:
Good idea!